src/software/ai/hl/stp/stp.cpp

```diff
@@ -168,8 +169,8 @@ std::vector<std::unique_ptr<Intent>> STP::getIntents(const World& world)
     return getIntentsFromCurrentPlay(world);
 }
 
-void STP::assignRobotsToTactics(const World& world,
-                                std::vector<std::shared_ptr<Tactic>> tactics) const
+std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
+    const World& world, std::vector<std::shared_ptr<Tactic>> tactics) const
```

Comment:
this should return a list of all tactics that were assigned to robots, including the goalie tactic and we need a regression test for this

Reply:
Done. I added a few tests to check if the returned tactics have expected values to those with goalie tactics. Please let me know if those suffice.